### ISSUE TITLE
Unable to connect to LG TV G3

### ISSUE THREAD

**Issue by VMmatty** [hobbyquaker/node-red-contrib-lgtv#58]
I doubt anyone is even looking at this anymore but just wanted to point out I'm unable to connect to the TV using this palette.  I have a 2023 LG G3 and while the native Home Assistant integration works and can do things like send text notifications, adjust volume, etc., the Node-Red palette can't seem to do it.  The configuration node is never able to connect to the LG TV whether I use a DNS name or an IP address.

It's strange that the Home Assistant integration is working without issue but this one doesn't.  Is anyone still maintaining this?  Hope so!

**Comment by vaibhav2912** [comment]
I have the same problem. Although I can see in history that mine was working until Oct 7th.

**Comment by VMmatty** [comment]
Sadly it's pretty clear this isn't being maintained anymore.

**Comment by Jibun-no-Kage** [comment]
Yeah, same issue for me as well, new 2023 50URA9000 series.  I did find a JS project that does work... but it is raw JS.  Here is the link to the working project for LG IP Control in JS.  [https://github.com/WesSouza/lgtv-ip-control](url).  You could establish a require statement in the settings,js for nodered, would not be as easy to use a a canned palette, but doable.

**Comment by hobbyquaker** [comment]
[node-red-contrib-lgtv 2.0.0](https://github.com/hobbyquaker/node-red-contrib-lgtv/releases/tag/v2.0.0) is released (npm `latest`); it needs Node-RED >= 4 and Node.js >= 20.19 (^20.19 || ^22.12 || >=24), see the [changelog](https://github.com/hobbyquaker/node-red-contrib-lgtv/blob/master/CHANGELOG.md).

TVs with firmware from 2023 on (and some older ones after updates) only accept the secure websocket on `wss://<tv>:3001`; 1.x only knew `ws://<tv>:3000`, which is why the connection failed or ended in `ECONNRESET`. 2.0 tries `wss:3001` first and falls back to `ws:3000` automatically (there is also a setting in the config node to pin one of them or another port). Please update to 2.0.0, click **Connect** in the config node again to pair, and reopen if the TV still does not connect - then a note which model/firmware you have would help.

(The Home Assistant integration switched to wss earlier, which is why it kept working. 2.0 does the same.)

---
_This comment was written by an AI agent (Claude Code) on behalf of @hobbyquaker._
